server/src/main/java/org/elasticsearch/threadpool/ThreadPool.java

```diff
@@ -321,6 +330,60 @@ public ThreadPool(final Settings settings, final ExecutorBuilder<?>... customBui
         this.cachedTimeThread.start();
     }
 
+    private static ArrayList<Instrument> setupMetrics(MeterRegistry meterRegistry, String name, ExecutorHolder holder) {
+        Map<String, Object> at = Map.of();
+        ArrayList<Instrument> instruments = new ArrayList<>();
+        if (holder.executor() instanceof ThreadPoolExecutor threadPoolExecutor) {
+            String prefix = "es.thread_pool." + name + ".threads.";
+            logger.info("Starting tp metrics {}", name);
+            instruments.add(
+                meterRegistry.registerLongGauge(
+                    prefix + "count",
+                    "number of threads for " + name,
+                    "count",
+                    () -> new LongWithAttributes(threadPoolExecutor.getPoolSize(), at)
+                )
+            );
+            instruments.add(
+                meterRegistry.registerLongGauge(
+                    prefix + "queue",
+                    "number queue size for " + name,
+                    "count",
+                    () -> new LongWithAttributes(threadPoolExecutor.getQueue().size(), at)
+                )
+            );
+            instruments.add(
+                meterRegistry.registerLongGauge(
+                    prefix + "active",
+                    "number of active threads for " + name,
+                    "count",
+                    () -> new LongWithAttributes(threadPoolExecutor.getActiveCount(), at)
+                )
+            );
+            instruments.add(
+                meterRegistry.registerLongGauge(
+                    prefix + "largest",
+                    "largest pool size for " + name,
+                    "count",
+                    () -> new LongWithAttributes(threadPoolExecutor.getLargestPoolSize(), at)
+                )
+            );
+            instruments.add(
+                meterRegistry.registerLongAsyncCounter(
+                    prefix + "completed",
```

Comment:
can we rename to "completed.total" ?

Reply:
Addressed in 1a0b1defba83ade4b673cf9af21fa09161397f50